int32 vi_idx # index of a vertex from which the edge stems
int32 vj_idx # index of a vertex to which the edge goes